lib/class-wp-theme-json-gutenberg.php

```diff
@@ -264,6 +264,14 @@ class WP_Theme_JSON_Gutenberg {
 		'filter' => array( 'filter', 'duotone' ),
 	);
 
+	/**
+	 * Protected style properties.
+	 *
+	 * These properties are only rendered if a corresponding setting in the
+	 * `PROPERTIES_METADATA` path enables it via a value other than `null`.
+	 */
+	const PROTECTED_PROPERTIES = array( '--wp--style--block-gap' );
```

Comment:
A small thing: Should we have something like `'spacing.blockGap' => 'spacing.blockGap'` instead. Meaning (theme.json style property mapping to its corresponding theme.json setting key). The current logic relies on the output instead of the input and assumes the setting key is the same as the style key (which is of course true now).